test/spec/directives/bsSwitchSpec.js

```diff
@@ -674,17 +675,16 @@ describe('Directive: bsSwitch', function () {
 
   function makeTestChange(input) {
     return function () {
-      compileDirective('change', input);
+      var element = compileDirective('change', input);
 
       scope.change = function() {};
       spyOn(scope, 'change');
 
-      scope.model = false;
-      scope.$apply();
-      expect(scope.change).not.toHaveBeenCalled();
+      element.find('input').click();
+      expect(scope.change).toHaveBeenCalled();
```

Comment:
This tests that the change function is called when the view is changed, which is ok, but we should also include a test for model changes.

I'd also include a `it` that tests if transitioning from and to an `undefined` state triggers the change fn.